tfhe/src/integer/server_key/radix_parallel/tests_signed.rs

```diff
@@ -2989,6 +3003,290 @@ where
+        let (tmp_ct, tmp_o) = sks.signed_overflowing_scalar_add_parallelized(&ctxt_0, clear_1);
+        assert!(ct_res.block_carries_are_empty());
+        assert_eq!(ct_res, tmp_ct, "Failed determinism check");
+        assert_eq!(tmp_o, result_overflowed, "Failed determinism check");
+
+        let (expected_result, expected_overflowed) =
+            signed_overflowing_add_under_modulus(clear_0, clear_1, modulus);
+
+        let decrypted_result: i64 = cks.decrypt_signed(&ct_res);
+        let decrypted_overflowed = cks.decrypt_bool(&result_overflowed);
+        assert_eq!(
+            decrypted_result, expected_result,
+            "Invalid result for add, for ({clear_0} + {clear_1}) % {modulus} \
+             expected {expected_result}, got {decrypted_result}"
+        );
+        assert_eq!(
+            decrypted_overflowed,
+            expected_overflowed,
+            "Invalid overflow flag result for overflowing_add for ({clear_0} + {clear_1}) % {modulus} \
+             expected overflow flag {expected_overflowed}, got {decrypted_overflowed}"
+        );
+        assert_eq!(result_overflowed.0.degree.get(), 1);
+        assert_eq!(result_overflowed.0.noise_level(), NoiseLevel::NOMINAL);
+
+        for _ in 0..NB_TESTS_SMALLER {
+            // Add non zero scalar to have non clean ciphertexts
+            let clear_2 = random_non_zero_value(&mut rng, modulus);
+            let clear_rhs = random_non_zero_value(&mut rng, modulus);
+
+            let ctxt_0 = sks.unchecked_scalar_add(&ctxt_0, clear_2);
+            let (clear_lhs, _) = signed_overflowing_add_under_modulus(clear_0, clear_2, modulus);
+            let d0: i64 = cks.decrypt_signed(&ctxt_0);
+            assert_eq!(d0, clear_lhs, "Failed sanity decryption check");
+
+            let (ct_res, result_overflowed) =
+                sks.signed_overflowing_scalar_add_parallelized(&ctxt_0, clear_rhs);
+            assert!(ct_res.block_carries_are_empty());
+            let (expected_result, expected_overflowed) =
+                signed_overflowing_add_under_modulus(clear_lhs, clear_rhs, modulus);
+
+            let decrypted_result: i64 = cks.decrypt_signed(&ct_res);
+            let decrypted_overflowed = cks.decrypt_bool(&result_overflowed);
+            assert_eq!(
+                decrypted_result, expected_result,
+                "Invalid result for add, for ({clear_lhs} + {clear_rhs}) % {modulus} \
+                expected {expected_result}, got {decrypted_result}"
+            );
+            assert_eq!(
+                decrypted_overflowed,
+                expected_overflowed,
+                "Invalid overflow flag result for overflowing_add, for ({clear_lhs} + {clear_rhs}) % {modulus} \
+                expected overflow flag {expected_overflowed}, got {decrypted_overflowed}"
+            );
+            assert_eq!(result_overflowed.0.degree.get(), 1);
+            assert_eq!(result_overflowed.0.noise_level(), NoiseLevel::NOMINAL);
+        }
+    }
+
```

Comment:
no hardcoded inputs to check that e.g. -modulus + -1 overflows ?

Reply:
We could add one

Reply:
adding a hardcoded test of  overflowing_sub(-1, -modulus) allowed to catch a little bug